src/general/base_stage/StageBase.cs

```diff
@@ -107,6 +113,15 @@ public override void _Process(float delta)
 
             wantsToSave = false;
         }
+
+        GameWorld.LightCycle.Process(delta);
```

Comment:
I think it would make sense to add a `GameWorld.Process` method that takes care of processing everything in the world.